sources/rest_api/config_setup.py

```diff
@@ -349,50 +350,96 @@ def _find_resolved_params(endpoint_config: Endpoint) -> List[ResolvedParam]:
     ]
 
 
-def _handle_response_actions(
-    response: Response, actions: List[ResponseAction]
-) -> Optional[str]:
-    """Handle response actions based on the response and the provided actions."""
-    content = response.text
-
-    for action in actions:
+def _action_type_unless_custom_hook(
+    action_type: Optional[str], custom_hook: Optional[List[Callable[..., Any]]]
+) -> Union[
+    Tuple[str, Optional[List[Callable[..., Any]]]],
+    Tuple[None, List[Callable[..., Any]]],
+]:
+    if custom_hook:
+        return (None, custom_hook)
+    return (action_type, None)
+
+
+def _handle_response_action(
+    response: Response, action: ResponseAction
+) -> Union[
+    Tuple[str, Optional[List[Callable[..., Any]]]],
+    Tuple[None, List[Callable[..., Any]]],
+    Tuple[None, None],
+]:
+    """
+    Checks, based on the response, if the provided action applies.
+    """
+    content: str = response.text
+    status_code = None
+    content_substr = None
+    action_type = None
+    custom_hook = None
+    response_action = None
+    if callable(action):
+        custom_hook = [action]
+    else:
+        action = cast(ResponseActionDict, action)
         status_code = action.get("status_code")
-        content_substr: str = action.get("content")
-        action_type: str = action.get("action")
+        content_substr = action.get("content")
+        response_action = action.get("action")
+        if isinstance(response_action, str):
+            action_type = response_action
+        elif callable(response_action):
+            custom_hook = [response_action]
+        elif isinstance(response_action, list) and all(
+            callable(action) for action in response_action
+        ):
+            custom_hook = response_action
```

Comment:
We probably can't replace these conditionals on runtime type checks with polymorphism – unless we require the user to pass in our custom classes which would be cumbersome for them.